### PR TITLE
Samples & readmes

### PR DESCRIPTION
**Pull request by gearama**
Samples and readmes for each sample, and the general readme. 
The general readme is not complete yet as i need to merge this to get the official links for various source files. 
fixed a couple of small bugs, and renames a type from RestoreDeletedSecretOperation to RecovereDeletedSecretOperation to match the operation name 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request.
+ * - AZURE_CLIENT_SECRET: The client secret.
+ *
+ */
+
+#if defined(_MSC_VER)
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <azure/core.hpp>
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp:20]: don't think we need this

File: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#include <azure/identity.hpp>
+#include <azure/keyvault/keyvault_secrets.hpp>
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+using namespace Azure::Security::KeyVault::Secrets;
+void AssertSecretsEqual(Secret const& expected, Secret const& actual);
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp:30]: This is not used in this sample. Should we remove it from here?

File: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample2-backup-restore/sample2-backup-restore.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    // get secret
+    Secret secret = secretClient.GetSecret(secretName).Value;
+
+    std::cout << "Secret is returned with name " << secret.Name << " and value " << secret.Value
+              << std::endl;
+
+    size_t backUpSize = 0;
+    {
+      std::cout << "\t-Backup Key" << std::endl;
+      std::vector<uint8_t> backupKey(secretClient.BackupSecret(secret.Name).Value.Secret);
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample2-backup-restore/sample2-backup-restore.cpp:61]:
```suggestion
      std::vector<uint8_t> backupSecret(secretClient.BackupSecret(secret.Name).Value.Secret);
```

mm,, if you ask me, this is a copy-construct that is not very usual.  We are discarding the entire response+model to only keep the Secret. Can we just do:

auto backupSecret = secretClient.BackupSecrect(secret.Name).Value;
auto const& secret = backupSecret.Secret;  

Then we are telling people we can keep the backupSecret in the heap and read the secret from it directly without making a copy.

File: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3-delete-recover.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    Secret secret = secretClient.GetSecret(secretName).Value;
+
+    std::cout << "Secret is returned with name " << secret.Name << " and value " << secret.Value
+              << std::endl;
+
+    // start deleting the secret
+    DeleteSecretOperation operation = secretClient.StartDeleteSecret(secret.Name);
+
+    // You only need to wait for completion if you want to purge or recover the secret.
+    operation.PollUntilDone(std::chrono::milliseconds(2000));
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3-delete-recover.cpp:62]: add `using namespace std::chrono_literals;` in the top and replace the `std::chrono::milliseconds(2000)` with just `2s` here.

Also, maybe do the polling on bigger jumps (like 1min).  For Keys, deleting will never take less than 30 seconds.  There's no benefit from pulling every less than that

**Review by vhvb1989 — COMMENTED** [review]
added some comments

File: sdk/keyvault/azure-security-keyvault-secrets/src/keyvault_operations.cpp [hunk 5]
```diff
@@ -9,9 +9,9 @@
 #include "azure/keyvault/secrets/keyvault_operations.hpp"
 #include "azure/keyvault/secrets/secret_client.hpp"
 #include "private/secret_serializers.hpp"
-// RestoreDeletedSecretOperation
+// RecoverDeletedSecretOperation
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/src/keyvault_operations.cpp:12]: it was there for my own personal tracking, to keep track of some things, i removed it

File: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp (hunk 1 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample1-basic-operations/sample1-basic-operations.cpp:20]: done and cleaned up all the other unused includes in all the other samples

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**